Makefile

```diff
@@ -1,14 +1,16 @@
 .DEFAULT_GOAL := help
 
 ARCH ?= `uname -m`
```

Comment:
I agree.

Reply:
I have done this. It took a little bit more work, because I realised that something had been broken a while back, and I needed to fix that. I have broken out those changes into two separate commits in this PR.

Reply:
I've looked at the change you are talking about and I can't say I follow it. I agree that the variables are used for two distinct purposes. Although this statement:

>Context 1. expects to look at the local environment at build time with
pg_config to determine what PG_VERSION is. It does not expect PG_VERSION
to be passed in from the outside.

was not accurate before `53a992f27f41e666331bbbd7fc87fbcd54085680`.

Furthermore, both sets of variables are altered when building artefacts: 
- `PG_BUILD_VERSION` and `PG_BUILD_VER` are used when building images. Although instead of passing them from the outside we set them in the Makefile itself.
- `PG_RELEASE_VERSION` and `PG_RELEASE_VER` are used for the packages and we pass them from the outside.

I suggest either better separate uses of build vs release variables (e.g. use build variables only with `cargo` and use release variables when building artefacts) or keep only one set of variables and make all of them overridable.